File: graphics/gl/src/main/java/io/github/aemogie/timble/gl/OpenGL.java
package io.github.aemogie.timble.gl;

import io.github.aemogie.timble.graphics.GraphicsAPI;
import io.github.aemogie.timble.graphics.Window;

import org.lwjgl.opengl.*;
import static org.lwjgl.glfw.GLFW.*;

public class OpenGL extends GraphicsAPI {
	
	@Override
	public boolean init(Window window) {
		glfwMakeContextCurrent(window.getGLFWWindowPointer());
		GL.createCapabilities();
		return true;
	}
}

File: graphics/gl/src/main/java/io/github/aemogie/timble/gl/OpenGL.kt
package io.github.aemogie.timble.gl

import io.github.aemogie.timble.graphics.Window
import org.lwjgl.glfw.GLFW.glfwMakeContextCurrent
import org.lwjgl.glfw.GLFW.glfwSwapBuffers
import org.lwjgl.opengl.GL

fun Window.useOpenGL() {
	subscribe(Window.InitEvent::class) {
		glfwMakeContextCurrent(windowPointer)
		GL.createCapabilities()
	}
	subscribe(Window.FrameLoopEvent::class) {
		glfwSwapBuffers(windowPointer)
	}
}